Include/internal/pycore_code.h

```diff
@@ -94,13 +85,32 @@ typedef struct {
 
 typedef struct {
     _Py_CODEUNIT counter;
-    _Py_CODEUNIT type_version;
-    _Py_CODEUNIT _t1;
+    _Py_CODEUNIT type_version[2];
     _Py_CODEUNIT func_version;
 } _PyBinarySubscrCache;
 
 #define INLINE_CACHE_ENTRIES_BINARY_SUBSCR CACHE_ENTRIES(_PyBinarySubscrCache)
 
+typedef struct {
+    _Py_CODEUNIT counter;
+    _Py_CODEUNIT version[2];
+    _Py_CODEUNIT index;
+} _PyAttrCache;
+
+#define INLINE_CACHE_ENTRIES_LOAD_ATTR CACHE_ENTRIES(_PyAttrCache)
+
+#define INLINE_CACHE_ENTRIES_STORE_ATTR CACHE_ENTRIES(_PyAttrCache)
+
+typedef struct {
+    _Py_CODEUNIT counter;
+    _Py_CODEUNIT type_version[2];
+    _Py_CODEUNIT dict_offset;
+    _Py_CODEUNIT keys_version[2];
+    _Py_CODEUNIT descr[4];
+} _PyLoadMethodCache;
```

Comment:
Given that `LOAD_METHOD_MODULE` and `LOAD_ATTR_MODULE` share code,  maybe change this to:
```
typedef struct {
     _PyAttrCache attr;
    _Py_CODEUNIT keys_version[2];
    _Py_CODEUNIT descr[4];
} _PyLoadMethodCache;
```

Reply:
I'd rather leave it as-is. Otherwise, accessing the `attr` methods, including the `counter`, is a bit awkward.

I don't think we're doing anything illegal here either, since we only use one consistent cache layout for each specialized form (meaning, `_PyLoadMethodCache` and `_AttrCache` members never alias each other). The exception is the `counter` member for *all* instructions, but C guarantees that it the first member of a struct will always be at offset `0`. So no issues there.